sherpa/models/model.py

```diff
@@ -800,6 +800,33 @@ def get_components_by_class(self, cls: type, subclass_ok: bool = True) -> list[M
         return parts
 
 
+    def __getitem__(self, key: Model | str) -> Model | list[Model]:
+        """Return the components of the model with the given name or type.
+
+        Parameters
```

Comment:
I think worth adding `.. versionchanged:: 4.17.1`  text here as it is a significant change.